samples/wfi32iot/basic-sample/src/iotconnect_app.c

```diff
@@ -202,30 +218,20 @@ static void publish_telemetry(int warming_up, int connections[]) {
     button_press_data.flag.sw1 = 0;
     button_press_data.flag.sw2 = 0;
     
+    
     //Air Quality 7 Click Reading
-    if(connections[0] == 1)
-    {
-        //Array entries are  [tVOC, CO2, Resistor Value, Status]
-        float air_data[] = {0, 0, 0, 0, 0, 0};
-        AIRQUALITY7_readData(air_data);
-        //If the sensor has been warming up for 15 minutes
-        if(warming_up == 181)
-        {
+    if(click_board_detection.air7 == true) {
+        air7_data_struct air7_data;
+        AIRQUALITY7_readData(&air7_data);
+        //If the sensor has been warming up for at least 15 minutes
+        if(running_time > 900) {
             //Report status and data from readData function output
-            if(air_data[3] == 1)
-            {
-               iotcl_telemetry_set_string(msg, "AIR7_Status", "ERROR"); 
-            }
-            else
-            {
-               iotcl_telemetry_set_string(msg, "AIR7_Status", "READY");
-            }
-            iotcl_telemetry_set_number(msg, "AIR7_tVOC_ppb", air_data[0]);   
-            iotcl_telemetry_set_number(msg, "AIR7_CO2_ppm", air_data[1]);
+            iotcl_telemetry_set_string(msg, "AIR7_Status", "READY");
+            iotcl_telemetry_set_number(msg, "AIR7_tVOC_ppb", air7_data.tvoc);   
+            iotcl_telemetry_set_number(msg, "AIR7_CO2_ppm", air7_data.co2);
         }
         //If the sensor is still warming up
-        else
-        {
+        else {
```

Comment:
Else still needs to be at the same line as the closing curly. You can move the commend inside the block.